### PR TITLE
Fix broken registration link in footer

### PR DESCRIPTION
**Pull request by haseleyi**
#### What? Why?

Closes [#3061](https://github.com/openfoodfoundation/openfoodnetwork/issues/3061)

We have a broken link: In the account page for a buyer account, the footer features a registration call partial that routes the user to /signup instead of /register. 
This fixes things. But I'm not satisfied with it: why would registration_path resolve to /signup on only this page (or at all for that matter)? I'd rather address the root of the problem.

#### What should we test?

In a buyer account, navigate to My Account > Orders and click "Register here" in the footer.

#### Release notes

Fixed a broken registration link in the footer of our My Account page.

Changelog Category: Fixed

### CODE REVIEW

**Comment by luisramos0** [comment]
good question: "why would registration_path resolve to /signup on only this page (or at all for that matter)?"
tricky one. Some clues:
this dependency: https://github.com/openfoodfoundation/openfoodnetwork/blob/1052d982550f0b7ac601b399f032291383ddad26/Gemfile#L18
takes you to this route:
https://github.com/openfoodfoundation/spree_auth_devise/blob/da9eecefc6fe13dedf4c6f3febec79caad839ec3/config/routes.rb#L18

Because the account page where you are is also on this routes file:
https://github.com/openfoodfoundation/spree_auth_devise/blob/da9eecefc6fe13dedf4c6f3febec79caad839ec3/config/routes.rb#L34
I think the registration_path defined on this routes file with /sign_up takes precedence the ofn's one in:
https://github.com/openfoodfoundation/openfoodnetwork/blob/1052d982550f0b7ac601b399f032291383ddad26/config/routes.rb#L19

There's also this devise_scope that I dont really know how it works.

I think we can consider this is a bug in OFN because we should not have re-written this registration named route. I dont think there is a good reason to do this but i am not 100% sure.

The proper fix could be to rename the registration path in the ofn context to separate it from the spree devise registration path.
Or maybe an easier fix would be to overwrite the devise registration path in ofn's routes.

Let's see what reviewers say.

**Comment by mkllnk** [comment]
Very good detective work @luisramos0. Most front-end pages are our own implementation, but we are still using Spree's UsersController and override the main action. I guess that Spree's controller is referring to Spree routes instead of OFN routes. So I think we could solve this problem by implementing our own account page. It seems out of scope, but is actually a fix we want to do anyway.

So we have some options now:

- Explicitly link `/register`.
- Rename our register path and use that.
- Overwrite Spree's register path.
- Add a route to redirect from `/signup` to `/register`.
- Move accounts page into the OFN namespace and abandon Spree's controller.

I'm surprised that there are so many solutions to this simple problem. I'm voting for the last one, but would accept the current pull request until that is implemented. A working link wins over code style. It would be nice to add a comment though. Otherwise someone could have the brilliant idea to change it back to `registration_path` again, because that's the right way to do it.

**Comment by luisramos0** [comment]
thanks @mkllnk 
I agree. That means we are approving this PR as long as a comment is added :-)

**Comment by haseleyi** [comment]
Thank you both for the excellent breakdown of the issue and potential solutions. This is a thoughtful and inspiring community to learn from!

File: app/views/shared/_register_call.html.haml [hunk 1]
```diff
@@ -1,6 +1,10 @@
 .alert-cta
   %h6
-    %a{href: registration_path, target: "_blank"}
+    -# Please forgive the hard-coded link:
+    -# The more elegant 'registration_path' resolves to /signup due to spree_auth_device > config > routes.rb
+    -# This is one of several possible fixes. Long-term, we'd like to bring the accounts page into OFN.
+    -# View the discussion here: https://github.com/openfoodfoundation/openfoodnetwork/pull/3174
```
Comment (sauloperez) [inline app/views/shared/_register_call.html.haml]: :ok_hand:

**Comment by RachL** [comment]
Deploying in AU staging (yes I jumped one issue, but I'm in a meeting in 15 minutes, so I figured rather I test 1 small one rather than nothing :) )

**Comment by RachL** [comment]
Link redirects correctly to https://staging.openfoodnetwork.org.au/register ! ready to go :)

**Pull request merged**